### ISSUE TITLE
Numerical Problem in Lennard-Jones Potential

### ISSUE THREAD

**Issue by WangXinyan940** [deepmodeling/DMFF#50]
I found that our implementation of Lennard-Jones potential has a numerical problem when eps=0 exists in the system.

The minimal reproduction is shown below:

```
import jax.numpy as jnp
import jax

sig_label = jnp.array([0, 1, 0, 2], dtype=int)
eps_label = jnp.array([0, 1, 0, 2], dtype=int)
sig_prm = jnp.array([1.0, 1.1, 1.0])
eps_prm = jnp.array([0.4, 0.5, 0.0])

pairs = jnp.array([
    [0, 1],
    [0, 2],
    [0, 3],
    [1, 2],
    [1, 3],
    [2, 3]
], dtype=int)

pos = jnp.array([
    [ 0.0, 0.0, 0.0],
    [ 0.5, 0.0, 0.0],
    [ 0.0, 0.5, 0.0],
    [ 0.2, 0.3, 0.1]
])

def getE(pos, sig_prm, eps_prm, sig_label, eps_label, pairs):
    sig_i = sig_prm[sig_label[pairs[:,0]]]
    sig_j = sig_prm[sig_label[pairs[:,1]]]
    eps_i = eps_prm[eps_label[pairs[:,0]]]
    eps_j = eps_prm[eps_label[pairs[:,1]]]

    eps = jnp.sqrt(eps_i * eps_j)
    sig = (sig_i + sig_j) * 0.5

    dr_vec = pos[pairs[:,0]] - pos[pairs[:,1]]

    dr_norm = jnp.linalg.norm(dr_vec, axis=1)
    dr_inv = 1.0 / dr_norm
    sig_dr = sig * dr_inv
    sig_dr6 = jnp.power(sig_dr, 6)
    sig_dr12 = jnp.power(sig_dr6, 2)
    E = 4.0 * eps * (sig_dr12 - sig_dr6)
    return jnp.sum(E)

print("GRAD of EPS -- eps_2 = 0")
print(jax.grad(getE, argnums=2)(pos, sig_prm, eps_prm, sig_label, eps_label, pairs))


eps_prm = jnp.array([0.4, 0.5, 1e-12])
print("GRAD of EPS -- eps_2 = 1e-12")
print(jax.grad(getE, argnums=2)(pos, sig_prm, eps_prm, sig_label, eps_label, pairs))
```

On my laptop, the first printed gradient is nan/inf when eps_2 is 0, while the gradient becomes normal when eps_2 is a tiny number.

**Comment by WangXinyan940** [comment]
Solved by using
```
eps = jnp.sqrt(eps_i * eps_j + 1e-12)
```
